url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/CCproxy.java

```diff
@@ -0,0 +1,159 @@
+package ccproxy;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import pl.cc.core.PauseType;
+
+public class CCproxy extends Thread {
+	static final String VERSION = "363";
+	static Parametry p = new Parametry();
+	static String version;
+	
+	org.apache.log4j.Logger log = p.log;
+
+	public void start(String configFile) throws IOException {
+		version="DEV";
+		InputStream vis=getClass().getResourceAsStream("/.version");
+		if (vis!=null) version=new BufferedReader(new InputStreamReader((vis))).readLine();
+		System.out.println("CCProxy version: "+version);
+		p.c = new ReadConfig(p, configFile);
+		ReadConfig c = p.c;
+		if (!checkLicense(p.c.licensedAgents, p.c.licenseKey, p.c.licenseExpires)){
+			System.out.println("License error");
+			System.exit(-1);
+		}
+		p.dbConn = new DBConnection(c.pghost, c.pgport, c.pgdbname, c.pguser, c.pgpass);
+		if (!c.configOk)
+			System.exit(-1);
+		loadPauseTypeList();
+		GadajAsterisk gadajAsteriskReader, gadajAsteriskWriter;
+		try {
+			gadajAsteriskReader = new GadajAsterisk(c.asteriskHost, c.asteriskPort,
+					c.asteriskUser, c.asteriskSecret, p);
+			p.gadajAsterisk = gadajAsteriskReader;
+			gadajAsteriskReader.start();
+			gadajAsteriskWriter = new GadajAsterisk();
+			gadajAsteriskWriter.start();
+			Listener listener = new Listener(c.listenPort, p);
+			listener.start();
+
+			while (true) {
+				sleep(1000);
+				if (gadajAsteriskReader.connected)
+					continue;
+				p.log.info("Stracono polaczenie z asterisk.");
+				System.exit(1);	// niestety na razie reconnect powoduje że nie przetwarzają się dalej wiadomości
+			}
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+
+	}
+
+	public CCproxy() throws IOException {
+	}
+
+	public CCproxy(String configFile) throws IOException {
+		start(configFile);
+	}
+
+	/**
+	 * Załadowanie z bazy danych listy definicji pauz
+	 */
+	private void loadPauseTypeList(){
+		log.debug("Loading pause definition list from database ...");
+		ResultSet rs = p.dbConn.query("select id_pause_def, name from t_pause_def;");
+
+		try {
+			while (rs.next()) {
+				PauseType pauseType = new PauseType(rs.getInt("id_pause_def"), rs.getString("name"));
+				p.pauseTypeList.add(pauseType);
+			}
+			p.pauseTypeList.sort();
+			log.info("Pause definition list loaded. Size: "+p.pauseTypeList.size());
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	static boolean checkLicense(String max_agents, String key, String expirydate) {
+
+		try {
+			File file = new File("/usr/sbin/dmidecode");
+			FileInputStream fis = new FileInputStream(file);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fis*. The resource is referenced at line 100. The resource is closed at line 102. There are other execution paths that do not contain closure statements, for example, when FileInputStream.read throws an exception. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).